src/Infrastructure.Styles.Analyzer/RMSYN1103ExtensionMethodsTestedAsNormalInvocationAnalyzer.cs

```diff
@@ -0,0 +1,112 @@
+
+        internal const string Category = "Readability";
+
+        private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId,
+            Title,
+            Message,
+            Category,
+            DiagnosticSeverity.Warning,
+            true,
+            Description);
+
+        private static readonly List<string> TestAttributeList = new List<string> { "Test", "TestAttribute" };
+
+        public override void Initialize (AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.EnableConcurrentExecution();
+
+            context.RegisterSyntaxNodeAction(AnalyzeInvocationExpression, SyntaxKind.InvocationExpression);
+        }
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+            ImmutableArray.Create(Rule);
+
+        private static void AnalyzeInvocationExpression (SyntaxNodeAnalysisContext context)
+        {
+            var invocationExpression = context.Node as InvocationExpressionSyntax;
+
+            if (!IsInTestMethod(invocationExpression!)) return;
+
+            var symbolInfo = context.SemanticModel.GetSymbolInfo(invocationExpression!, context.CancellationToken);
+            if (symbolInfo.Symbol is not IMethodSymbol { IsExtensionMethod: true } methodSymbol)
+                return;
+
+            if (methodSymbol.ReducedFrom == null)
+                return;
+
+            context.ReportDiagnostic(Diagnostic.Create(Rule, context.Node.GetLocation()));
+        }
+
+        private static bool IsInTestMethod (InvocationExpressionSyntax invocationExpression)
+        {
+            if (invocationExpression.Parent is not ExpressionStatementSyntax
+                { Parent: BlockSyntax
+                    { Parent: MethodDeclarationSyntax parentMethod }
+                })
+                return false;
+
+            if (parentMethod.AttributeLists.Count == 0)
+                return false;
+
+            foreach (var attributes in parentMethod.AttributeLists.SelectMany(attributeList => attributeList.Attributes))
+            {
+                if (attributes.Name.IsKind(SyntaxKind.IdentifierToken))
+                    if (IsContainedInTestAttributeList(attributes.Name.ToString()))
+                        return true;
+
+                if (attributes.Name.IsKind(SyntaxKind.IdentifierName))
+                    if (IsContainedInTestAttributeList(attributes.Name.ToString()))
+                        return true;
+
+                if (attributes.Name is QualifiedNameSyntax name)
+                    if (IsContainedInTestAttributeList(name.Right.ToString()))
+                        return true;
+            }
+            return false;
+        }
+
+        private static bool IsContainedInTestAttributeList (string identifier)
+        {
+            return TestAttributeList.TrueForAll(a => a.StartsWith(identifier));
```

Comment:
I think we want `.Any` here -- this will fail a test if we have an `[TestAttribute]` if I am not mistaken.